### PR TITLE
Move fastSquareRoot to Math folder

### PR DESCRIPTION
**Pull request by duncanspumpkin**
fastSquareRoot is used by a large amount of functions for calculating the distance between two locations.

### CODE REVIEW

File: src/OpenLoco/Math/Vector.hpp [hunk 1]
```diff
@@ -270,4 +270,12 @@ namespace OpenLoco::Math::Vector
         };
     }
 
+    uint16_t fastSquareRoot(uint32_t distance);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Math/Vector.hpp:273]: This looks a bit awkward. Perhaps add some extra whitespace to make it more obvious what's going on?

```suggestion
    uint16_t fastSquareRoot(uint32_t distance);

```

File: src/OpenLoco/Vehicles/VehicleBody.cpp [hunk 2]
```diff
@@ -268,62 +265,51 @@ namespace OpenLoco::Vehicles
         if (object_sprite_type == 0xFF)
             return;
 
-        auto distance_x = front_bogie->x - back_bogie->x;
-        auto distance_y = front_bogie->y - back_bogie->y;
-
-        auto offset = fastSquareRoot(distance_x * distance_x + distance_y * distance_y);
+        auto xDiff = front_bogie->x - back_bogie->x;
+        auto yDiff = front_bogie->y - back_bogie->y;
+        auto distanceBetweenBogies = Math::Vector::distance(Map::map_pos{ front_bogie->x, front_bogie->y }, Map::map_pos{ back_bogie->x, back_bogie->y });
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Vehicles/VehicleBody.cpp:270]: I think it would be clearer if these parameters were assigned to local variables.

Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/VehicleBody.cpp:270]: hmm perhaps if we had a method that returned a map_pos from the base struct.

Comment (AaronVanGeffen) [reply src/OpenLoco/Vehicles/VehicleBody.cpp:268]: That would be even better. If you're planning something like that already in a follow-up, I'm fine with this as-is.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**